src/Regions.py

```diff
@@ -62,6 +64,12 @@ def create_region(world: World, multiworld: MultiWorld, player: int, name: str,
             loc_id = world.location_name_to_id.get(location, 0)
             locationObj = ManualLocation(player, location, loc_id, ret)
             ret.locations.append(locationObj)
+            if location in DataValidation.location_events_table:
+                eventLocationObj = ManualLocation(player,"["+location+"]",None,ret)
+                eventLocationObj.show_in_spoiler = False
+                eventItemOjb = ManualItem("["+location+"]",ItemClassification.progression, None, player)
```

Comment:
I think this explains why you wanted to go with `|[whatever]|` for the format in requires. But we don't want to present locations to the user as "special items". We want to present them as locations, as their own thing. This is consistent with how we present requirement functions differently, too.

For example, if we use this in a requires string:
```
"requires": "|Key| and |Shield| and [Saved Zelda in the Castle]"
```
... then it is visually clear that "Saved Zelda in the Castle" isn't an item. Because, to the user / world creator, it isn't. It's a location. We don't need to visually represent it in the same way that we have to for making it functional. 

In the background, we can spice it up with extra `[]` or `__LOCATION__:` or whatever we want the format of the event to be. So that `Saved Zelda in the Castle` actually matches `THISISALOCATION:Saved Zelda in the Castle` or whatever.

For an example of how we'd handle the `[]` delimiter like `||`, check out Rules.py around line 82 or so. That's the implementation for handling the `{}` delimiter for requirement functions, and you could probably borrow a lot of what it's doing.

Reply:
The problem is that as an event, as far as AP is concerned it **IS** just an item, the only difference is it's an item that doesn't get sent to the user, or even live beyond the scope of generation

Like i can make the change so || is validated as an item, and [] is validated as a location, but under the hood they're both turned into item access rules, if that satisfies your requirement that they look different to builders and players

Reply:
Yep, exactly. That change is what I'm suggesting. 😃 

Manual was originally designed for -- and still caters to a lot of -- less technical world creators. A lot of these users will never see the internals of AP (or even a custom apworld), so they won't see the correlation. To them, they just see (via Manual) a locations.json and an items.json and fill in stuff, so the two have separate meaning for them. Manual already changes the presentation from the implementation in some cases to reduce complexity for the user, so that's okay (and desirable) to do here, too. Also avoids an event discussion that doesn't need to happen with the user, because it's not information they need. 😃